paramiko/pipe.py

```diff
@@ -46,25 +47,50 @@ def __init__(self):
         self._closed = False
 
     def close(self):
-        os.close(self._rfd)
-        os.close(self._wfd)
-        # used for unit tests:
         self._closed = True
 
+        # We cannot do anything about closing errors. It is only a
+        # "best effort" approach
+        try:
+            os.close(self._rfd)
+        except:
```

Comment:
Best practice is to catch `Exception` and subclasses, or else you get `BaseException` and subclasses, which would also trap things like `KeyboardInterrupt`.

```suggestion
        except Exception:
```